File: py/src/practise150/p13_container_with_most_water_test.py
from src.practise150.p13_container_with_most_water import Solution

def test_max_area():
    test_cases = [
        ([1,7,2,5,4,7,3,6], 36),
        ([2,2,2],4),
        ([1,5,4,3], 6),
        ([3,1,2,4,5],12),
        ([2,1,8,6,4,6,5,5],25),
    ]
    s = Solution()
    for test_case in test_cases:
        result = s.maxArea(test_case[0])
        assert result == test_case[1]

File: go/practise150/p13_container_with_most_water_test.go
package practise150

import (
	"fmt"
	"testing"
)

func TestMaxAmountOfWater(t *testing.T) {
	type TestCases struct {
		input    []int
		expected int
	}

	test_cases := []TestCases{
		{
			input:    []int{1, 7, 2, 5, 4, 7, 3, 6},
			expected: 36,
		},
		{
			input:    []int{12, 2, 2},
			expected: 4,
		},
		{
			input:    []int{1, 5, 4, 3},
			expected: 6,
		},
		{
			input:    []int{3, 1, 2, 4, 5},
			expected: 12,
		},
		{
			input:    []int{2, 1, 8, 6, 4, 6, 5, 5},
			expected: 25,
		},
	}

	for idx, test_case := range test_cases {
		t.Run(fmt.Sprintf("Test_case_no:%d", idx+1), func(t *testing.T) {
			actual := MaxAmountOfWater(test_case.input)
			if actual != test_case.expected {
				t.Errorf("Testing Max amount of water. Input: %v, Expected: %d, Actual: %d", test_case.input, test_case.expected, actual)
			}
		})
	}
}
